Library/ViewModels/ConfirmDetailsViewModel.swift

```diff
@@ -0,0 +1,264 @@
+     */
+    let nonLocalPickupShippingLocationViewHidden = Signal.combineLatest(baseReward, rewards, context)
+      .map { baseReward, rewards, context in
+        [
+          context.shippingLocationViewHidden,
+          !baseReward.shipping.enabled,
+          rewards.count > 1
+        ].contains(true)
+      }
+
+    self.shippingLocationViewHidden = Signal
+      .combineLatest(nonLocalPickupShippingLocationViewHidden, baseReward)
+      .map { flag, baseReward in
+        isRewardLocalPickup(baseReward) ? true : flag
+      }
+
+    /**
+     Shipping summary view is hidden when updating,
+     if the base reward has no shipping, when NO add-ons were selected or when base reward has local pickup option.
+     */
+    let nonLocalPickupShippingSummaryViewHidden = Signal.combineLatest(baseReward, rewards, context)
+      .map { baseReward, rewards, context in
+        [
+          context.isAny(of: .update, .changePaymentMethod, .fixPaymentMethod),
+          !baseReward.shipping.enabled,
+          rewards.count == 1
+        ].contains(true)
+      }
+
+    self.shippingSummaryViewHidden = Signal.combineLatest(nonLocalPickupShippingSummaryViewHidden, baseReward)
+      .map { flag, baseReward in
+        isRewardLocalPickup(baseReward) ? true : flag
+      }
+
+    let shippingViewsHidden: Signal<Bool, Never> = Signal.combineLatest(
+      self.shippingSummaryViewHidden,
+      self.shippingLocationViewHidden
+    )
+    .map { a, b -> Bool in
+      let r = a && b
+      return r
+    }
+
+    self.localPickupViewHidden = baseReward.map(isRewardLocalPickup).negate()
+
+    // Only shown for regular non-add-ons based rewards
+    self.configureShippingLocationViewWithData = Signal.combineLatest(
+      projectAndReward,
+      shippingViewsHidden.filter(isFalse),
+      selectedLocationId
+    )
+    .map { projectAndReward, _, selectedLocationId in
+      (projectAndReward.0, projectAndReward.1, selectedLocationId)
+    }
+    .map { project, reward, locationId in
+      (project, reward, true, locationId)
+    }
+
+    // Only shown for add-ons based rewards
+    self.configureShippingSummaryViewWithData = Signal.combineLatest(
+      selectedShippingRule.skipNil().map(\.location.localizedName),
+      project.map(\.stats.omitUSCurrencyCode),
+      project.map { project in
+        projectCountry(forCurrency: project.stats.currency) ?? project.country
+      },
+      allRewardsShippingTotal
+    )
+    .map(PledgeShippingSummaryViewData.init)
+
+    self.configurePledgeAmountViewWithData = Signal.combineLatest(
+      projectAndReward,
+      initialAdditionalPledgeAmount
+    )
+    .map(unpack)
+    .map { project, reward, additionalPledgeAmount in
+      (
+        project,
+        reward,
+        additionalPledgeAmount
+      )
+    }
+
+    // Only shown for if the shipping summary view and shipping location view are hidden
+    self.configureLocalPickupViewWithData = Signal.combineLatest(
```

Comment:
Nit: Seems like a lot of this VM is for local pickup/shipping issues. Do we do any kind of view/vm composition for issues like this? Feels like it could be logically separate, in some ways, but I'm not sure I've seen us doing that elsewhere in the codebase.